examples/ssh/macos_disk_partition.js

```diff
@@ -0,0 +1,205 @@
+      if (match) {
+        var key = match[1].trim()
+        var value = match[2].trim()
+        if (key === 'Device Identifier') {
+          if (Object.keys(currentDisk).length > 0) {
+            partitions.push(currentDisk)
+          }
+          currentDisk = { 'Device Identifier': value }
+        } else {
+          if (!currentDisk) { continue }
+          currentDisk[key] = value
+        }
+      }
+    }
+    if (Object.keys(currentDisk).length > 0) {
+      partitions.push(currentDisk)
+    }
+    var filteredDisks = partitions.filter(function(disk) {
+      return disk['Mount Point'] && disk['Mount Point'].includes('/Volumes')
+    })
+    return filteredDisks
+
+  } else {
+    console.error('No data found')
+    D.failure(D.errorType.PARSING_ERROR)
+  }
+}
+
+function sanitize (output) {
+  var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history']
+  var recordIdSanitizationRegex = new RegExp(recordIdReservedWords.join('|'), 'g')
+  return output.replace(recordIdSanitizationRegex, '').slice(0, 50).replace(/\s+/g, '-').toLowerCase()
+}
+
+function convertToAppropriateUnits(bytes) {
+  if (bytes >= 1e9) {
+    return {
+      value: (bytes / 1e9).toFixed(2),
+      unit: 'GB'
+    }
+  } else {
+    return {
+      value: (bytes / 1e6).toFixed(2),
+      unit: 'MB'
+    }
+  }
+}
+
+/**
+ * Extracts relevant variables from parsed disk partitons and populates the table
+ * @param {Array} partitions Array of objects representing parsed disk partitonsinformation
+ */
+function populateTable(partitions) {
+  var parentDiskValues = {}
+  var table
+  partitions.forEach(function (disk) {
+    var containerTotalSpace = disk['Container Total Space'] ? disk['Container Total Space'].match(/(\d+) Bytes/) : 0
+    var volumeTotalSpace = disk['Volume Total Space'] ? disk['Volume Total Space'].match(/(\d+) Bytes/) : 0;
+    var totalSpaceBytes = containerTotalSpace ? parseInt(containerTotalSpace[1]) : (volumeTotalSpace ? parseInt(volumeTotalSpace[1]) : 0)
+    var containerFreeSpace = disk['Container Free Space'] ? disk['Container Free Space'].match(/(\d+) Bytes/) : 0
+    var volumeFreeSpace = disk['Volume Free Space'] ? disk['Volume Free Space'].match(/(\d+) Bytes/) : 0
+    var totalFreeSpaceBytes = containerFreeSpace ? parseInt(containerFreeSpace[1]) : (volumeFreeSpace ? parseInt(volumeFreeSpace[1]) : 0)
+    var totalUsedSpace = totalSpaceBytes - totalFreeSpaceBytes
+    var percentageUsedSpace = totalSpaceBytes > 0 ? (totalUsedSpace / totalSpaceBytes) * 100 : 0
+    var parentDisk = disk['Part of Whole']
+    if (!parentDiskValues[parentDisk]) {
+      var freeSpace = convertToAppropriateUnits(totalFreeSpaceBytes)
+      var size = convertToAppropriateUnits(totalSpaceBytes)
+      parentDiskValues[parentDisk] = {
+        freeSpace: freeSpace.value,
+        size: size.value,
+        freeSpaceUnit: freeSpace.unit,
+        sizeUnit: size.unit
+      }
+      if (!table) {
+        table = D.createTable(
```

Comment:
please move the table creation outside of the function instead of checking if it has been created